### PR TITLE
Shorten Fix longest C++ lines longer than 100 chars

### PR DESCRIPTION
**Pull request by tkruse**
Shortening the code lines is important for several alternative kinds of display (PDF, print, mobile phone), because long code lines either crosses some right margin, or code is wrapped like text which is horrible to read.

Maybe an 80 chars limit would be even better, but choosing a slightly smaller font should give acceptable results as well.

### CODE REVIEW

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -2237,10 +2240,11 @@ Passing a shared smart pointer (e.g., `std::shared_ptr`) implies a run-time cost
 ##### Example
 
     void f(int*);             // accepts any int*
-    void g(unique_ptr<int>);  // can only accept ints for which you want to transfer ownership
-    void g(shared_ptr<int>);  // can only accept ints for which you are willing to share ownership
+    void g(unique_ptr<int>);  // accepts ints to transfer ownership
```
Comment (blakehawkins) [inline CppCoreGuidelines.md]: Are we not concerned that some meaning is lost here?

File: CppCoreGuidelines.md [hunk 2]
```diff
@@ -2663,10 +2681,14 @@ A `span` represents a range of elements, but how do we manipulate elements of th
 
     void f(span<int> s)
     {
-        for (int x : s) cout << x << '\n';  // range traversal (guaranteed correct)
-        for (int i = 0; i < s.size(); ++i) cout << x << '\n';  // C-style traversal (potentially checked)
-        s[7] = 9;                           // random access (potentially checked)
-        std::sort(&s[0], &s[s.size() / 2]);    // extract pointers (potentially checked)
+        // range traversal (guaranteed correct)
+        for (int x : s) cout << x << '\n';
```
Comment (blakehawkins) [inline CppCoreGuidelines.md]: I think for sequences of commented lines, some extra white space might be good, i.e.:

    // range traversal (guaranteed correct)
    for (int x : s) cout << x << '\n';
    
    // C-style traversal (potentially checked)
    for (int i = 0; i < s.size(); ++i) cout << x << '\n';

It makes it more clear which lines the comments are paired with.

File: CppCoreGuidelines.md [hunk 3]
```diff
@@ -4035,7 +4064,7 @@ If the `Handle` owns the object referred to by `s` it must have a destructor.
 
 Independently of whether `Handle` owns its `Shape`, we must consider the default copy operations suspect:
 
-    Handle x {*new Circle{p1, 17}};  // the Handle had better own the Circle or we have a leak
+    Handle x {*new Circle{p1, 17}};  // causes a leak if the Handle is not a Circle
```
Comment (blakehawkins) [inline CppCoreGuidelines.md]: Similar to above, some of the meaning has changed. The old comment discussed ownership. Maybe it's better this way but worth thinking about.

**Comment by tkruse** [comment]
fair enough, I made the suggested changes

**Comment by AndrewPardoe** [comment]
Thank you!

**Pull request merged**